### PR TITLE
Clean BuildFiles in Validation Subsystem

### PR DESCRIPTION
**Pull request by guitargeek**
#### PR description:

Another quick partially automatic BuildFile cleaning PR in the style of many before (for example https://github.com/cms-sw/cmssw/pull/29295), this time covering the full Validation subsystem.

The changes mostly stay trivial but for the moving a usage of the `TYPELOOKUP_DATA_REG` macro to the file where it should be used according to [1] (quoting here and I hope I understood it right):
```
 The actual name of the file that uses the 'TYPELOOKUP_DATA_REG' macro is not important.  The only important point
 the file that uses the 'TYPELOOKUP_DATA_REG' macro must be in the same library as the data class it is registering.
```

Doing this correctly, one does not have to declare the library where the macro is actually used as an additional dependency anymore if one wants to use a `edm::ESHandle<TrackerTopology>`, which would look like a fake dependency to this cleaning because it is not required by any `include` statement.

[1] https://github.com/cms-sw/cmssw/blob/master/FWCore/Utilities/interface/typelookup.h#L34

#### PR validation:

CMSSW compiles. It was checked that all newly added dependencies are actually required by the package with `git grep`.

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by cvuosalo** [comment]
+1

**Comment by jfernan2** [comment]
+dqm

**Comment by slava77** [comment]
@cmsbuild please test

to get cleaner diffs (expected starting from 07-2300 IB)

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/29404	62962fc
- related to reco: ```TYPELOOKUP_DATA_REG(TrackerTopology)``` moved to DataFormats/TrackerCommon. This looks OK, even though here the use case is not really obvious

File: DataFormats/TrackerCommon/src/TrackerTopology.cc [hunk 1]
```diff
@@ -447,3 +448,5 @@ int TrackerTopology::getITPixelLayerNumber(const DetId &id) const {
   }
   return layer;
 }
+
+TYPELOOKUP_DATA_REG(TrackerTopology);
```
Comment (makortel) [inline DataFormats/TrackerCommon/src/TrackerTopology.cc:452]: This should actually go into `ES_TrackerTopology.cc`, see https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToRegisterESData

Comment (slava77) [reply DataFormats/TrackerCommon/src/TrackerTopology.cc:452]: thank you, I'm taking off reco +1

**Comment by guitargeek** [comment]
Okay I hope I got it right with the most recent commit, @makortel!

**Comment by makortel** [comment]
> Okay I hope I got it right with the most recent commit, @makortel!

Yes, thanks!

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by jfernan2** [comment]
+1

**Comment by cvuosalo** [comment]
+1

**Comment by qliphy** [comment]
+1

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/29404	c0ad81b
- since the last (attempt to) signoff for reco, the ```TYPELOOKUP_DATA_REG``` was now correctly placed in DataFormats/TrackerCommon/src/ES_TrackerTopology.cc

**Comment by rekovic** [comment]
+1

**Comment by guitargeek** [comment]
Hi @makortel, can you sign for core at this point or do you have further comments? Thank you!

**Comment by makortel** [comment]
+1

(sorry, somehow missed core signature was required)

**Comment by silviodonato** [comment]
+1

**Pull request merged**